### ISSUE TITLE
ConnectionPool hangs in thin mode when timeout is not None

### ISSUE THREAD

**Issue by lpetre-ulb** [oracle/python-oracledb#166]
Hello,

while porting my project to use a `ConnectionPool` instead of many `Connection`, I stumbled across an issue when:
* using the thin mode;
* setting the timeout to an integer (so not `None`);
* and using a dynamic number of connections (i.e. `min` != `max`).

In this specific case, the `ConnectionPool.acquire()` hangs and never returns a connection. I could identify the problem as due to to the following while loop. I'm not familiar enough with the library intrinsics to provide a PR though. https://github.com/oracle/python-oracledb/blob/7eb9be4bfed94663d03880967ac725e24972593b/src/oracledb/impl/thin/pool.pyx#L329-L333

The project uses Python 3.8.13 with `oracledb` 1.3.1. The problem is however present since the feature implementation in https://github.com/oracle/python-oracledb/commit/c7c0d55e1381282d73a131f923379d6ce1605123.

Please find a minimal reproducible example below. The script blocks and never completes, staying in an endless loop in `ConnectionPool.acquire()`.

``` python
import oracledb
from multiprocessing.pool import ThreadPool

pool = oracledb.create_pool("<REDACTED>", config_dir="/etc", min=1, max=5, timeout=10)
thread_pool = ThreadPool(processes = 5)

def wait(i):
  with pool.acquire() as connection:
    with connection.cursor() as cursor:
      cursor.execute("""
        BEGIN
          DBMS_SESSION.sleep(1);
        END; """)
      print(f"Done {i}")

for i in range(10):
  thread_pool.apply_async(wait, args=[i])
```

Thank you for any help. Let me know if you need more informations.

**Comment by lpetre-ulb** [comment]
Wonderful! Thank you for your prompt answer.

**Comment by anthony-tuininga** [comment]
I have pushed a patch that should correct this issue. If you are able to [build from source](https://python-oracledb.readthedocs.io/en/latest/user_guide/installation.html#installing-from-source-code) you can verify that it corrects your issue as well.

**Comment by anthony-tuininga** [comment]
This has been included in version 1.3.1 which was just released!

**Comment by lpetre-ulb** [comment]
Hello, all my apologies for the delayed reply. I confirm that the bug is indeed fixed in version 1.3.1. Thank you for the patch and the release.
